core/src/services/azblob/backend.rs

```diff
@@ -545,7 +548,17 @@ impl Access for AzblobBackend {
         let status = resp.status();
 
         match status {
-            StatusCode::OK => parse_into_metadata(path, resp.headers()).map(RpStat::new),
+            StatusCode::OK => {
+                let headers = resp.headers();
+                let mut meta = parse_into_metadata(path, headers)?;
+
+                let user_meta = parse_prefixed_headers(&headers, X_MS_META_NAME_PREFIX);
```

Comment:
Should the metadata parsing be done in two steps (first call `parse_into_metadata` and then `parse_prefixed_headers`) or should we add a new `prefix: Optional<&str>` parameter to the `parse_into_metadata` function and the prefixed headers parsing will be done there (and also the `meta.with_user_metadata` call, as the `parse_into_metadata` returns a `Metadata` struct)

Reply:
Hi, I tend to keep `parse_into_metadata` simple because not all services handle metadata and user metadata in the same way.

As I mentioned before, OpenDAL tends to have more readable code. It's acceptable for us to have some duplicated code here and there. We can revisit this part when we have more similar code, rather than just two or three instances.